src/insertion_sort.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""The following code implements an insertion_sort function designed
+to be used on a standart Python list."""
+from __future__ import unicode_literals
+import random
+import datetime
+
+
+def insertion_sort(sort_lst):
+    if not isinstance(sort_lst, list):
+        raise TypeError("Your input must be a Python list.")
+    sort_position = -1
+    while sort_position < len(sort_lst):
```

Comment:
You're basically doing the same thing as a for loop here, so it would easier to read and understand as a for loop over the indices and/or values in `sort_list`.